packages/react/src/hooks/__tests__/useHeightDimensions.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import { useHeightDimension } from '../useHeightDimension';
+import { renderHook, act } from '../../../../../tests/test-utils';
+import React from 'react';
+
+const observeMock = jest.fn();
+const unobserveMock = jest.fn();
+const disconnectMock = jest.fn();
+
+global.ResizeObserver = jest.fn(() => ({
+  observe: observeMock,
+  unobserve: unobserveMock,
+  disconnect: disconnectMock,
+})) as unknown as typeof ResizeObserver;
+describe('useHeightDimension', () => {
+  beforeEach(() => {
+    observeMock.mockClear();
+    unobserveMock.mockClear();
+    disconnectMock.mockClear();
+  });
+  it('should initialize height as 0px when isExpanded is false', () => {
+    const { result } = renderHook(() => useHeightDimension(false));
+
+    expect(result.current.height).toBe('0px');
+  });
+
+  it('should update height to contentRef.scrollHeight when isExpanded is true', () => {
+    const mockScrollHeight = 100;
+    const mockDiv = document.createElement('div');
+    Object.defineProperty(mockDiv, 'scrollHeight', { value: mockScrollHeight });
+
+    const { result, rerender } = renderHook(
+      ({ isExpanded }: { isExpanded: boolean }) =>
+        useHeightDimension(isExpanded),
+      {
+        initialProps: { isExpanded: false },
+      },
+    );
+
+    act(() => {
```

Comment:
I think `act` is deprecated by the testing-library. You can change to this approach:

```suggestion
    await waitFor(() => ...)
```